### PR TITLE
CreateEventFromVOBJ: if there is no dtend, use the duration if available

### PR DESCRIPTION
**Pull request by tschwinge**
... which allows me to `gcalcli import` the `*.ics` files provided by a travel agency.  According to a quick skim of one random ;-) iCalendar RFC, that appears to be legit.

Beware: only very lightly tested, so far.

### CODE REVIEW

**Comment by jcrowgey** [comment]
Would you be willing to rebase off current master and then do a bit more testing?   I think it would be good to add your calendar file to the tests/data/ directory and set up some sort of unit testing for your code?

File: gcalcli/gcal.py [hunk 1]
```diff
@@ -1424,25 +1424,17 @@ def CreateEventFromVOBJ(ve):
...
-                    print('End..........%s' % ve.dtend.value.isoformat())
-                if ve.dtstart.value:
-                    print('Local Start..%s' %
-                          self._localize_datetime(ve.dtstart.value)
-                          )
-                if ve.dtend.value:
-                    print('Local End....%s' %
-                          self._localize_datetime(ve.dtend.value)
-                          )
+            #if not hasattr(ve, 'dtend') and not hasattr(ve, 'duration'):
```
Comment (jcrowgey) [inline gcalcli/gcal.py]: I don't think we want to merge changes that comment out blocks.  We should just delete unused code.

File: gcalcli/gcal.py [hunk 2]
```diff
@@ -1476,17 +1471,39 @@ def CreateEventFromVOBJ(ve):
...
+                    # calculation on.  TODO: correct?
+                    if isinstance(ve.dtstart.value, datetime):
+                        event['end'] = {'dateTime': end.isoformat(),
+                                        'timeZone': self.cals[0]['timeZone']}
+                    else:
+                        event['end'] = {'date': end.isoformat()}
+
                 else:
                     event['end'] = event['start']
+                #DebugPrint('event['start']: %s\n' % event['start'])
```
Comment (jcrowgey) [inline gcalcli/gcal.py]: Let's remove these comments too.

File: gcalcli/gcal.py [hunk 3]
```diff
@@ -1476,17 +1471,39 @@ def CreateEventFromVOBJ(ve):
...
 
+                elif hasattr(ve, 'duration') and ve.duration.value:
+                    if verbose:
+                        print('Duration.....%s' % ve.duration.value)
+                    end = ve.dtstart.value + ve.duration.value
+                    if verbose:
+                        print('Calculated End........%s' % end.isoformat())
+                        print('Calculated Local End..%s' % self._localize_datetime(end))
+                    # Decide based on dtstart; that's what we base our
+                    # calculation on.  TODO: correct?
```
Comment (jcrowgey) [inline gcalcli/gcal.py]: Can you follow up on this TODO?

**Comment by tschwinge** [comment]
@jcrowgey: Thanks for looking into this, and no worries about the delay.  I also haven't really been proactive with *ping*ing this.

Thanks for your review; that all makes sense, and I'll re-work/extend as suggested.  I won't get it done this year, but I'll try to before the "third birthday" of this PR.  ;-)

**Comment by dbarnett** [comment]
> I'll try to before the "third birthday" of this PR.

Now 7 years old! Do you still have interest in this change or would you prefer to abandon and let someone else pick it up later?

**Comment by dbarnett** [comment]
K, I rebased a version of these changes onto the latest main in #771, doing some polishing and then getting that merged.

**Pull request closed without merging**